### PR TITLE
[FLINK-15562][docs] Add Example settings.xml for maven archetype command wh…

### PR DESCRIPTION
**Pull request by KarmaGYZ**
…en using Maven 3.0 or higher

## What is the purpose of the change

Add Example settings.xml for maven archetype command when using Maven 3.0 or higher. 

## Brief change log

- Add Example settings.xml for maven archetype command when using Maven 3.0 or higher.
- Remove the redundant intent and blank lines in the corresponding docs.

After this change:
image

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no

### CODE REVIEW

File: docs/dev/projectsetup/scala_api_quickstart.md [hunk 1]
```diff
@@ -128,27 +128,46 @@ Use one of the following commands to __create a project__:
...
 {% endhighlight %}
     </div>
     {% unless site.is_stable %}
     <p style="border-radius: 5px; padding: 5px" class="bg-danger">
-        <b>Note</b>: For Maven 3.0 or higher, it is no longer possible to specify the repository (-DarchetypeCatalog) via the commandline. If you wish to use the snapshot repository, you need to add a repository entry to your settings.xml. For details about this change, please refer to <a href="http://maven.apache.org/archetype/maven-archetype-plugin/archetype-repository.html">Maven official document</a>
+        <b>Note</b>: For Maven 3.0 or higher, it is no longer possible to specify the repository (-DarchetypeCatalog) via the command line. For details about this change, please refer to <a href="http://maven.apache.org/archetype/maven-archetype-plugin/archetype-repository.html">Maven official document</a>
+        If you wish to use the snapshot repository, you need to add a repository entry to your settings.xml. For example:
+{% highlight bash %}
+<settings>
+  <activeProfiles>
```
Comment (zentol) [inline docs/dev/projectsetup/scala_api_quickstart.md]: what's with the profile stuff?

File: docs/dev/projectsetup/scala_api_quickstart.zh.md [hunk 2]
```diff
@@ -126,27 +126,47 @@ addSbtPlugin("com.typesafe.sbteclipse" % "sbteclipse-plugin" % "4.0.0")
 
 <div class="tab-content">
     <div class="tab-pane active" id="maven-archetype">
-    {% highlight bash %}
-    $ mvn archetype:generate                               \
-      -DarchetypeGroupId=org.apache.flink              \
-      -DarchetypeArtifactId=flink-quickstart-scala     \{% unless site.is_stable %}
```
Comment (zentol) [inline docs/dev/projectsetup/scala_api_quickstart.zh.md]: let's move these refactorings into a separate commit

File: docs/dev/projectsetup/java_api_quickstart.md [hunk 3]
```diff
@@ -53,28 +53,47 @@ Use one of the following commands to __create a project__:
...
 {% endif %}
-    {% endhighlight %}
+{% endhighlight %}
 
     </div>
     {% unless site.is_stable %}
     <p style="border-radius: 5px; padding: 5px" class="bg-danger">
-        <b>Note</b>: For Maven 3.0 or higher, it is no longer possible to specify the repository (-DarchetypeCatalog) via the command line. If you wish to use the snapshot repository, you need to add a repository entry to your settings.xml. For details about this change, please refer to <a href="http://maven.apache.org/archetype/maven-archetype-plugin/archetype-repository.html">Maven official document</a>
+        <b>Note</b>: For Maven 3.0 or higher, it is no longer possible to specify the repository (-DarchetypeCatalog) via the command line. For details about this change, please refer to <a href="http://maven.apache.org/archetype/maven-archetype-plugin/archetype-repository.html">Maven official document</a>
+        If you wish to use the snapshot repository, you need to add a repository entry to your settings.xml. For example:
```
Comment (zentol) [inline docs/dev/projectsetup/java_api_quickstart.md]: snapshots should at no point be advertised to users. They purely exist for dev purposes.

**Comment by KarmaGYZ** [comment]
Thanks for the review and information @zentol . If it is against ASF rules. I agree to not give that advice. I'm not sure should we also remove the existing note.

BTW, I'd like to keep those format refactor. WDYT?

**Comment by zentol** [comment]
Ah hold on, this is only shown in the documentation for master; it's probably fine in there.

File: docs/dev/projectsetup/scala_api_quickstart.md (hunk 1 above)
Comment (KarmaGYZ) [reply docs/dev/projectsetup/scala_api_quickstart.md]: Sorry, I did not understand what you mean. Do you mean to set it as the default profile?

Comment (zentol) [reply docs/dev/projectsetup/scala_api_quickstart.md]: I'm wondering why there is a profile at all?

Comment (zentol) [reply docs/dev/projectsetup/scala_api_quickstart.md]: In other words, why not like this:
```
<settings>
  <repositories>
    <repository>
      <id>apache-snapshots</id>
      <url>https://repository.apache.org/content/repositories/snapshots/</url>
    </repository>
  </repositories>
</settings>
```

I suppose the profile allows users to separate options per project; what I'm wondering is whether we shouldn't just document the minimum required contents?

... it's probably fine as is, just wondering, really.

**Comment by zentol** [comment]
@KarmaGYZ If you rebase the PR and move the reformatting into a separate commit we can proceed with the merge. Apologies for the confusion I caused!

File: docs/dev/projectsetup/scala_api_quickstart.md (hunk 1 above)
Comment (KarmaGYZ) [reply docs/dev/projectsetup/scala_api_quickstart.md]: I'm not a master for maven at all. But I think the current way follows the official schema. https://maven.apache.org/settings.html

**Comment by KarmaGYZ** [comment]
Thanks for the review @zentol . I'm not an expert for maven, any advice is appreciated. I've split the PR.
